src/vs/platform/extensionManagement/common/extensionGalleryService.ts

```diff
@@ -645,6 +649,12 @@ export class ExtensionGalleryService implements IExtensionGalleryService {
 					}
 				});
 			}
+
+			// {{SQL CARBON EDIT}} - filter out extensions that match the excludeFlags options
+			const flags = query.criteria.filter(x => x.filterType === FilterType.ExcludeWithFlags).map(v => v.value ? v.value.toLocaleLowerCase() : undefined);
```

Comment:
You're lower casing the exclude flags but it doesn't look like that's being done for the actual flags themselves